### ISSUE TITLE
`dart fix` should accept `--enable-experiment` option like `dart format` 

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/sdk#54142]
I was trying out the `extension type`s yesterday for the first time. I tried both in `flutter channel beta` and `flutter channel main` and the bug still persists (more details below about the specific versions).

I enabled `- inline-class` inside my `analysis_options.yaml` and the inconsistency I got was that neither `dart fix` nor `dart format` automatically could by themselves fix the formating inside my `extension type`. 

Only `dart format --enable-experiment=inline-class` was able to fix formatting for it taking into consideration my line length of 80 (tested with that).

The project I used for testing this is [here](https://github.com/FMorschel/dart_flutter_project_manager) and the file is under [lib/src/classes/dart_directory.dart](https://github.com/FMorschel/dart_flutter_project_manager/commit/4907e6d61168d95f155a0ff06e57d719d97a40ff#diff-358486012cc12c90e00cb4669554f0662fe3296f9394ae02d5a3f1e164801212) (Initial commit).

`flutter doctor -v` in main:

```
[√] Flutter (Channel main, 3.17.0-15.0.pre.9, on Microsoft Windows [Version 10.0.22621.2715], locale pt-BR)
    • Flutter version 3.17.0-15.0.pre.9 on channel main at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c7d4b32fdd (15 hours ago), 2023-11-22 12:16:37 -0800
    • Engine revision 342fb003b1
    • Dart version 3.3.0 (build 3.3.0-152.0.dev)
    • DevTools version 2.30.0-dev.4

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\Users\felip_0vh5fa6\AppData\Local\Android\sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: C:\Users\felip_0vh5fa6\AppData\Local\Programs\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.4.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.4.33403.182
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2022.3)
    • Android Studio at C:\Users\felip_0vh5fa6\AppData\Local\Programs\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)

[√] VS Code (version 1.84.2)
    • VS Code at C:\Users\felip_0vh5fa6\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.74.0

[√] VS Code (version 1.79.0-insider)
    • VS Code at C:\Users\felip_0vh5fa6\AppData\Local\Programs\Microsoft VS Code Insiders
    • Flutter extension version 3.64.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.2715]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 119.0.6045.160
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 119.0.2151.44

[√] Network resources
    • All expected network resources are available.

• No issues found!
```

`dart info` in main:

```
#### General info

- Dart 3.3.0-152.0.dev (dev) (Mon Nov 20 12:03:48 2023 -0800) on "windows_x64"
- on windows / "Windows 10 Pro" 10.0 (Build 22621)
- locale is pt-BR

#### Project info

- sdk constraint: '>=3.3.0-0 <4.0.0'
- dependencies: collection, equatable, filesystem_picker, flutter, flutterando_metrics, gap, git, go_router, path, process_run, provider, shared_preferences
- dev_dependencies: flutter_lints, flutter_test
```

`flutter doctor -v` in beta:

```
[√] Flutter (Channel beta, 3.17.0-0.0.pre, on Microsoft Windows [Version 10.0.22621.2715], locale pt-BR)
    • Flutter version 3.17.0-0.0.pre on channel beta at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 0b591f2c82 (3 weeks ago), 2023-11-02 22:01:10 -0400
    • Engine revision dff66925dc
    • Dart version 3.3.0 (build 3.3.0-91.0.dev)
    • DevTools version 2.29.0

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\Users\felip_0vh5fa6\AppData\Local\Android\sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: C:\Users\felip_0vh5fa6\AppData\Local\Programs\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.4.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.4.33403.182
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2022.3)
    • Android Studio at C:\Users\felip_0vh5fa6\AppData\Local\Programs\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)

[√] VS Code (version 1.84.2)
    • VS Code at C:\Users\felip_0vh5fa6\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.74.0

[√] VS Code (version 1.79.0-insider)
    • VS Code at C:\Users\felip_0vh5fa6\AppData\Local\Programs\Microsoft VS Code Insiders
    • Flutter extension version 3.64.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.2715]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 119.0.6045.160
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 119.0.2151.44

[√] Network resources
    • All expected network resources are available.

• No issues found!
```

`dart info` in beta:

```
#### General info

- Dart 3.3.0-91.0.dev (dev) (Thu Nov 2 09:10:03 2023 -0700) on "windows_x64"
- on windows / "Windows 10 Pro" 10.0 (Build 22621)
- locale is pt-BR

#### Project info

- sdk constraint: '>=3.3.0-0 <4.0.0'
- dependencies: collection, equatable, filesystem_picker, flutter, flutterando_metrics, gap, git, go_router, path, process_run, provider, shared_preferences
- dev_dependencies: flutter_lints, flutter_test
```

**Comment by lrhn** [comment]
By default, every tools should have the experiment supplied when you use it, none of them should just enable by default.
That's intended. 

That the analyzer can have it enabled by a configuration file is special, because the analysis server is not something you run on the command line.
And tools running through the analysis servers could probably use the same flags. That's https://github.com/dart-lang/sdk/issues/54139

**Comment by FMorschel** [comment]
But there is no option to enable this on `dart fix`, or at least I could not find it. Only `dart format` does. Then this should be taken into consideration if my previous statement is correct.

**Comment by lrhn** [comment]
That too should work. Dart-fix should accept the experiment flags, and pass them onto the analyzer code it's build on.

**Comment by FMorschel** [comment]
When I run `dart fix --enable-experiment=inline-class`:

```
Could not find an option named "enable-experiment".

Usage: dart fix [arguments]
-h, --help                      Print this usage information.
-n, --dry-run                   Preview the proposed changes but make no changes.
    --apply                     Apply the proposed changes.
    --code=<code1,code2,...>    Apply fixes for one (or more) diagnostic codes.

Run "dart help" to see global options.
```

When I run `dart help`

```
A command-line utility for Dart development.

Usage: dart <command|dart-file> [arguments]

Global options:
-v, --verbose               Show additional command output.
    --version               Print the Dart SDK version.
    --enable-analytics      Enable analytics.
    --disable-analytics     Disable analytics.
    --suppress-analytics    Disallow analytics for this `dart *` run without changing the analytics configuration.
-h, --help                  Print this usage information.

Available commands:
  analyze    Analyze Dart code in a directory.
  compile    Compile Dart to various formats.
  create     Create a new Dart project.
  devtools   Open DevTools (optionally connecting to an existing application).
  doc        Generate API documentation for Dart projects.
  fix        Apply automated fixes to Dart source code.
  format     Idiomatically format Dart source code.
  info       Show diagnostic information about the installed tooling.
  pub        Work with packages.
  run        Run a Dart program.
  test       Run tests for a project.

Run "dart help <command>" for more information about a command.
See https://dart.dev/tools/dart-tool for detailed documentation.
```

**Comment by FMorschel** [comment]
Will this change be available in the stable channel on the 3.3.0 release?
